mycity/mycity/intents/get_alerts_intent.py

```diff
@@ -77,27 +77,69 @@ def get_alerts_intent(
+    :param alerts_to_speech_output_function_for_test: Injectable function
     for unit tests
     :return: MyCityResponseDataModel object
     """
     logger.debug('MyCityRequestDataModel received:' +
                  mycity_request.get_logger_string())
 
-    alerts = get_alerts() if get_alerts_function_for_test is None \
-        else get_alerts_function_for_test()
-    logger.debug("[dictionary with alerts scraped from boston.gov]:\n" +
-                 str(alerts))
+    # get the intent_variables and sessions_attribute object from the request
+    intent_variables = mycity_request.intent_variables
+    session_attributes = mycity_request.session_attributes
+    decision = intent_variables['Decision'].get('value') \
+        if 'Decision' in intent_variables else None
+    session_alerts = session_attributes.get('alerts')
 
-    pruned_alerts = prune_normal_responses(alerts) \
-        if prune_normal_responses_function_for_test is None \
-        else prune_normal_responses_function_for_test(alerts)
-    logger.debug("[dictionary after pruning]:\n" + str(alerts))
+    logger.debug('decision: ' + str(decision) +
+                 ', session_alerts: ' + str(session_alerts))
 
+    # Build the response.
+    #   - if decision was made before asking alert, remind to ask for alert
+    #   - if there is no decision - probably asking for alerts, list alerts
+    #   - if there is decision w/ alerts, get the appropriate alert
     mycity_response = _create_response_object()
-    mycity_response.output_speech = alerts_to_speech_output(pruned_alerts) \
-        if alerts_to_speech_output_function_for_test is None \
-        else alerts_to_speech_output_function_for_test(pruned_alerts)
+    mycity_response.session_attributes = session_attributes.copy()
+
+    if decision is None:
+        alerts = get_alerts() if get_alerts_function_for_test is None \
+            else get_alerts_function_for_test()
+        logger.debug("[dictionary with alerts scraped from boston.gov]:\n" +
+                     str(alerts))
+
+        pruned_alerts = prune_normal_responses(alerts) \
+            if prune_normal_responses_function_for_test is None \
+            else prune_normal_responses_function_for_test(alerts)
+        logger.debug("[dictionary after pruning]:\n" + str(pruned_alerts))
+        pruned_alerts = { k.lower(): v for k, v in pruned_alerts.items() }
+
+        if len(pruned_alerts) > 1:
+            mycity_response.session_attributes['alerts'] = pruned_alerts
+            mycity_response.should_end_session = False
+            mycity_response.output_speech = list_alerts_output(pruned_alerts)
+        else:
+            mycity_response.output_speech = \
+                alerts_to_speech_output(pruned_alerts) \
+                if alerts_to_speech_output_function_for_test is None \
+                else alerts_to_speech_output_function_for_test(pruned_alerts)
+    elif session_alerts is None:
+        mycity_response.should_end_session = False
```

Comment:
Is it possible to have no saved `session_alerts` but have a `decision`? I wasn't able to reach this code with my testing.